tests/unit/s2n_connection_test.c

```diff
@@ -162,5 +163,88 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(server_conn));
     }
 
+    /* Test: get selected digest alg */
+    {
+        struct s2n_connection *conn;
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+
+        s2n_handshake_hash_algorithm output = { 0 };
+
+        EXPECT_FAILURE(s2n_connection_get_selected_digest_algorithm(NULL, &output));
+        EXPECT_FAILURE(s2n_connection_get_selected_digest_algorithm(conn, NULL));
+        EXPECT_FAILURE(s2n_connection_get_selected_client_cert_digest_algorithm(NULL, &output));
+        EXPECT_FAILURE(s2n_connection_get_selected_client_cert_digest_algorithm(conn, NULL));
+
+        EXPECT_SUCCESS(s2n_connection_get_selected_client_cert_digest_algorithm(conn, &output));
+        EXPECT_EQUAL(S2N_HANDSHAKE_HASH_NONE, output);
+
+        EXPECT_SUCCESS(s2n_connection_get_selected_digest_algorithm(conn, &output));
+        EXPECT_EQUAL(S2N_HANDSHAKE_HASH_NONE, output);
+
+        s2n_hash_algorithm inputs[] = { S2N_HASH_MD5, S2N_HASH_SHA1, S2N_HASH_SHA224, S2N_HASH_SHA256, S2N_HASH_SHA384, 
+                                        S2N_HASH_SHA512, S2N_HASH_SENTINEL, S2N_HASH_NONE };
+        EXPECT_EQUAL(S2N_HASH_SENTINEL, s2n_array_len(inputs));
+
+        s2n_handshake_hash_algorithm expected_output[] = { S2N_HANDSHAKE_HASH_MD5, S2N_HANDSHAKE_HASH_SHA1, 
+                                                           S2N_HANDSHAKE_HASH_SHA224, S2N_HANDSHAKE_HASH_SHA256, 
+                                                           S2N_HANDSHAKE_HASH_SHA384, S2N_HANDSHAKE_HASH_SHA512, 
+                                                           S2N_HANDSHAKE_HASH_NONE, S2N_HANDSHAKE_HASH_NONE };
+        EXPECT_EQUAL(s2n_array_len(expected_output), s2n_array_len(inputs));
+
+        for (int i = 0; i < s2n_array_len(inputs); i++) {
```

Comment:
We generally prefer size_t over int when iterating to s2n_array_len

Reply:
Okay thank you